src/routes/Diagram.ts

```diff
@@ -98,49 +97,32 @@ export class MermaidRoute extends OpenAPIRoute {
   }
 
   /// 3. Handles the API request
-  async handle(request: Request, env: Env, _ctx, data: Record<string, any>) {
+  async handle(request: Request, env: Env, _ctx : unknown, data: Record<string, any>) {
     const BASE_URL = new URL(request.url).origin
     const timeline = new Timeline();
 
     // Extract data from request
-    const diagramLanguage = 
+    const diagramLanguage =
       new URL(request.url).searchParams.get("diagramLanguage") as DiagramLanguage
       ?? "mermaid"; // For older versions
-    const diagramParam = 
+    const diagramParam =
       new URL(request.url).searchParams.get("diagram")
-      ?? new URL(request.url).searchParams.get("mermaid"); // For older versions
+      ?? new URL(request.url).searchParams.get("mermaid") as string; // For older versions
 
     const topic  = new URL(request.url).searchParams.get("topic") ?? "none";
-    const diagramType  = new URL(request.url).searchParams.get("diagramType")
-      ?? diagramParam.split('\n')[0]
+    const diagramType  = new URL(request.url).searchParams.get("diagramType") as DiagramType
       ?? "unknown";
+
     console.log('diagram', diagramParam)
     console.log('topic', topic)
 
-    // Print headers
+    const diagram = await diagramDetails(diagramParam, diagramLanguage)
+
     const headers = Object.fromEntries(request.headers)
     console.log('headers', headers)
 
-    const conversationId = headers['openai-conversation-id']
-    const ephemeralUserId = headers['openai-ephemeral-user-id']
-    const realIP = headers['x-real-ip']
+    const track = getTrack(headers, env)
 
-    // Track render event
-    const track = async (event: string, properties: Record<string, any>) => {
-      try {
-        const adjustedProperties = {
-          ...properties,
-          'conversation_id': conversationId,
-
-          ip: realIP,
-          'worker_environment': env.WORKER_ENV,
-        }
-        await sendMixpanelEvent(env.MIXPANEL_TOKEN as string, event, ephemeralUserId, adjustedProperties)
-        console.log('Sent mixpanel event', event, properties)
-      } catch (e) {
-        console.log('Error sending mixpanel event', e)
-      }
-    }
```

Comment:
Great minds think alike ahahahah. See on main, I have also moved this to mixpanel helper function

Reply:
Haha, indeed!